### PR TITLE
fix: Add server keep alive enforcement policy

### PR DESCRIPTION
**Pull request by sakoush**
**What this PR does / why we need it**:

This PR adds the relevant parameters to make client and server side keep alive work nice together. We should not allow these parameters to be set independently but in some cases they exist in different packages, so manual care is required.

This PR also tidy up some exponential backoff settings.

**Which issue(s) this PR fixes**:

### CODE REVIEW

File: scheduler/pkg/agent/modelserver_controlplane/oip/v2.go [hunk 1]
```diff
@@ -51,19 +50,12 @@ func CreateV2GrpcConnection(v2Config V2Config) (*grpc.ClientConn, error) {
 		grpc_retry.WithMax(v2Config.GRPRetryMaxCount),
 	}
 
-	kacp := keepalive.ClientParameters{
```
Comment (lc525) [inline scheduler/pkg/agent/modelserver_controlplane/oip/v2.go:54]: could we add a note on why the keepalive was removed here?

File: scheduler/pkg/util/grpc.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+func GetServerKeepAliveEnforcementPolicy() keepalive.EnforcementPolicy {
+	return keepalive.EnforcementPolicy{
+		MinTime:             gRPCKeepAliveTime,
+		PermitWithoutStream: gRPCKeepAlivePermit,
+	}
+}
+
+func GetClientExponentialBackoff() *backoff.ExponentialBackOff {
+	backOffExp := backoff.NewExponentialBackOff()
+	backOffExp.MaxElapsedTime = 0 // Never stop due to large time between calls
```
Comment (lc525) [inline scheduler/pkg/util/grpc.go:36]: do we want to move these to constants as well?

File: scheduler/pkg/kafka/gateway/worker.go [hunk 3]
```diff
@@ -124,12 +123,6 @@ func (iw *InferWorker) getGrpcClient(host string, port int) (v2.GRPCInferenceSer
 		creds = insecure.NewCredentials()
 	}
 
-	kacp := keepalive.ClientParameters{
```
[the remark this answers is not in the document]
Comment (sakoush) [reply scheduler/pkg/kafka/gateway/worker.go:127]: this is a grpc unary call to envoy for to perform inferences, not relevant yet for keepalive settings.

File: scheduler/pkg/agent/modelserver_controlplane/oip/v2.go (hunk 1 above)
Comment (sakoush) [reply scheduler/pkg/agent/modelserver_controlplane/oip/v2.go:54]: agent and model server are on the same pod/network and therefore no need to set keepalive

**Review by lc525 — APPROVED** [review]
lgtm, just some minor nits.

File: scheduler/pkg/util/grpc.go (hunk 2 above)
Comment (sakoush) [reply scheduler/pkg/util/grpc.go:36]: done in [bb639c1](https://github.com/SeldonIO/seldon-core/pull/6016/commits/bb639c1dc845bae35279249628fe8817978e80df)

**Pull request merged**